internal/btf/btf_test.go

```diff
@@ -2,47 +2,44 @@ package btf
 
 import (
 	"bytes"
-	"compress/gzip"
 	"encoding/binary"
 	"errors"
 	"fmt"
 	"io"
 	"os"
+	"sync"
 	"testing"
 
 	"github.com/cilium/ebpf/internal"
 	"github.com/cilium/ebpf/internal/testutils"
 )
 
-func parseVmLinux(tb testing.TB) (*Spec, error) {
-	fh, err := os.Open("testdata/vmlinux-btf.gz")
-	if err != nil {
-		tb.Fatal(err)
-	}
-	defer fh.Close()
+var vmlinux struct {
+	sync.Once
+	err error
+	raw []byte
+}
 
-	rd, err := gzip.NewReader(fh)
-	if err != nil {
-		tb.Fatal(err)
-	}
+func readVmLinux(tb testing.TB) *bytes.Reader {
```

Comment:
This is a drive by refactoring, I can split it into a separate commit.

Reply:
Sure, would be nice if it's not too difficult.